dpnp/dpnp_utils/dpnp_utils_linearalgebra.py

```diff
@@ -777,22 +849,35 @@ def dpnp_matmul(
         # behaves differently for matmul and vecdot
         order = "C"
 
+    if axis is not None:
+        signature, distinct_core = _get_signature(func)
+        raise TypeError(
+            f"{func}: axis can only be used with a single shared core "
+            f"dimension, not with the {distinct_core} distinct ones implied "
+            f"by signature {signature}."
+        )
```

Comment:
I guess the check is missing:
```suggestion
        if distinct_core != 1:
            raise TypeError(
                f"{func}: axis can only be used with a single shared core "
                f"dimension, not with the {distinct_core} distinct ones implied "
                f"by signature {signature}."
            )
```
Is that covered well in the test?

Reply:
This section of the code is executed only for "matmul," "matvec," and "vecmat," which always have more than one distinct core. In other words, if the `axis` parameter is provided to these functions, they will always raise an error. Adding `if distinct_core != 1:` correctly enforces this behavior but creates an issue in the coverage report since it's impossible to write a test where `distinct_core == 1` and execution reaches this part of the code.
I added a comment here to clarify it.

Reply:
We can also add an assert `distinct_core != 1`.